misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/RangeFormatter.java

```diff
@@ -0,0 +1,138 @@
+        for (int[] position : positions) {
+            LinePosition startPos = LinePosition.from(position[0], position[1]);
+            LinePosition endPos = LinePosition.from(position[2], position[3]);
+            LineRange lineRange = LineRange.from(null, startPos, endPos);
+            syntaxTree = Formatter.format(syntaxTree, lineRange);
+        }
+        Assert.assertEquals(syntaxTree.toSourceCode(), getSourceText(assertFilePath));
+    }
+
+    /**
+     * Test the formatting functionality for parser test cases.
+     *
+     * @param sourcePath Source path of the parser test
+     */
+    public void testParserResources(String sourcePath) throws IOException, FormatterException {
+        Path filePath = Paths.get(sourcePath);
+        String content = getSourceText(filePath);
+        TextDocument textDocument = TextDocuments.from(content);
+        SyntaxTree syntaxTree = SyntaxTree.from(textDocument);
+        if (!syntaxTree.hasDiagnostics()) {
+            SyntaxTree newSyntaxTree = Formatter.format(syntaxTree);
+            Assert.assertEquals(newSyntaxTree.toSourceCode(), getSourceText(filePath));
+        }
+    }
+
+    /**
+     * Defines the data provider object for test execution.
+     *
+     * @return Data provider for tests
+     */
+    @DataProvider(name = "test-file-provider")
+    public abstract Object[][] dataProvider();
+
+    /**
+     * List of file names configured to be skipped during the test execution.
+     *
+     * @return Skipped test file list
+     */
+    public List<String> skipList() {
+        return new ArrayList<>();
+    }
+
+    /**
+     * Specify the file names to be tested during the test execution.
+     *
+     * @return Test scenarios for execution
+     */
+    public Object[][] testSubset() {
+        return new Object[0][];
+    }
+
+    /**
+     * Returns the directory path inside resources which holds the test files.
+     *
+     * @return Directory path of test files
+     */
+    public abstract String getTestResourceDir();
+
+    protected Object[][] getConfigsList() {
+        if (this.testSubset().length != 0) {
+            return this.testSubset();
+        }
+        List<String> skippedTests = this.skipList();
+        try {
+            return Files.walk(this.resourceDirectory.resolve(this.getTestResourceDir()).resolve(ASSERT_DIR))
+                    .filter(path -> {
+                        File file = path.toFile();
+                        return file.isFile() && file.getName().endsWith(".bal")
+                                && !skippedTests.contains(file.getName());
+                    })
+                    .map(path -> new Object[]{path.toFile().getName(), this.getTestResourceDir()})
+                    .toArray(size -> new Object[size][2]);
+        } catch (IOException e) {
+            return new Object[0][];
+        }
+    }
+
+    private String getSourceText(Path sourceFilePath) throws IOException {
```

Comment:
should we rename this to `getFileContent`, since it reads both sources and asserts?